scripts/notarize-macos-dmg.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Copyright (C) 2024-2024	The DOSBox Staging Team
+
+# This script notarizes a macOS DMG produced from the CI builds by
+# extracting the DMG, signing the app bundle, using the rest of the files
+# and the newly-signed app bundle to create a new DMG, then signing and 
+# notarizing the entire new DMG.
+
+# Ensure developer identity and keychain profile name are set as environment variables
+# For example:
+# export DEVELOPER_IDENTITY="Developer ID Application: Valued Developer (xxxxxxxxxxx)"
+# export KEYCHAIN_PROFILE="Valued Developer Personal"
+
+if [[ -z "$DEVELOPER_IDENTITY" || -z "$KEYCHAIN_PROFILE" ]]; then
+    echo "Error: DEVELOPER_IDENTITY and KEYCHAIN_PROFILE environment variables must be set."
+    exit 1
+fi
+
+if [ "$#" -ne 2 ]; then
+    echo "Usage: $0 <input_dmg_path> <output_directory>"
+    exit 1
+fi
+
+SCRIPT_DIR=$(pwd)
+INPUT_DMG="$1"
+OUTPUT_DIR="$2"
```

Comment:
Can't we just do `OUTPUT_DMG="$2"` and get rid of the OUTPUT_DIR business? The output is just a single file.

Reply:
Right now it uses the input file name and adds -notarized to it, and the output just specify the target directory, rather than the entire output DMG path - but, could change it to do that instead.

Reply:
Yeah, I think INFILE OUTFILE is pretty standard.